src/rotation_generator.jl

```diff
@@ -246,3 +246,30 @@ function Base.show(io::IO, ::MIME"text/plain", X::RotationGenerator)
     io = IOContext(io, :typeinfo => eltype(X))
     Base.print_array(io, X)
 end
+
+"""
+    isrotationgenerator(r)
+    isrotationgenerator(r, tol)
+
+Check whether `r` is a rotation generator matrix (skew-symmetric matrix).
+"""
+isrotationgenerator
+
+function isrotationgenerator(r::StaticMatrix{N,N,T}) where {N,T<:Real}
+    m = SMatrix(r)
+    return iszero(m+m')
```

Comment:
Could alternatively be `≈(-m, m', atol=..., rtol=..., norm=...)` and reuse the `atol` from `≈` (also `rtol` if necessary?)  I believe this uses the 2-norm by default though and that might need to be customized.

Reply:
Do we need the `isapprox` here? I thought the usage of `isrotationgenerator` will be similar to `issymmetric`.
```julia
julia> using LinearAlgebra

julia> issymmetric([1.0 0.1;0.1 1.0])
true

julia> issymmetric([1.0 nextfloat(0.1);0.1 1.0])  # no tol options
false
```
Rotation matrices are not able to be represented precisely with floating-point numbers, but rotation generator matrix (skew-symmetric) matrices can.

Reply:
I don't know if this is necessary; the `tol` version in the docs triggered this thought and might be required if the skew symmetric matrix is constructed numerically.

But I'm also happy with no tolerance to follow `issymmetric` like you mentioned.

In that case it would be nice to use a version which does a minimum of arithmetic, roughly `m == -m'` but incremental as in the `ishermitian(::AbstractMatrix)` implementation, at least for non-StaticArray.